### ISSUE TITLE
waitForFileChooser() never resolves

### ISSUE THREAD

**Issue by fredck** [puppeteer/puppeteer#5537]
### Steps to reproduce

```plaintext
// By calling waitForFileChooser(), the file chooser dialog doesn't show up.
// I'm not sure this is expected.
let fileChooser = page.waitForFileChooser();

// This works properly, because the file chooser shows up if I comment the above line.
page.click( '#fc-issue_body' );

// This promise never resolves.
fileChooser = await fileChooser;

// This line is never reached.
fileChooser.accept( [ 'test.png' ] );
```

**Tell us about your environment:**

*   Puppeteer version: 2.1.1 (built-in Chromium)
*   Platform / OS version: macOS 10.15.3
*   Node.js version:13.11.0

**What is the expected result?**

The file chooser should accept the provided file.

**What happens instead?**

`waitForFileChooser()` never resolves and the code gets blocked.

**Comment by mlovatonv** [comment]
I think you have to call both functions at the same time, like the documentation says https://github.com/puppeteer/puppeteer/blob/master/docs/api.md#pagewaitforfilechooseroptions

```
const [fileChooser] = await Promise.all([
  page.waitForFileChooser(),
  page.click('#upload-file-button'), // some button that triggers file selection
]);
await fileChooser.accept(['/tmp/myfile.pdf']);
```

Also, this function seems to be more stable in puppeteer v2.0.0

**Comment by fredck** [comment]
> I think you have to call both functions at the same time, like the documentation says https://github.com/puppeteer/puppeteer/blob/master/docs/api.md#pagewaitforfilechooseroptions

I've tried that way and it doesn't work either. Same results.

Actually, I believe that the documentation is mistaken (that's why my implementation is different) because `Promise.all()` doesn't guarantee the order of execution of the promises in the array (maybe). Anyway, both implementation have the very same effect.

**Comment by mlovatonv** [comment]
I would recommend you trying with Chromium 79.0.3942.0 - Puppeteer v2.0.0 for now because I have seen some issues related to the fileChooser in the newer versions of pptr. This is because of the `page.handleFileChooser` removal (https://chromium.googlesource.com/chromium/src/+/41361e490363590c21250dfd94554f8200e041f1). I am not sure if this has been fixed, it wasn't working for me when I tried with pptr v2.1.0 and v2.1.1.

**Comment by fredck** [comment]
> I would recommend you trying with Chromium 79.0.3942.0 - Puppeteer v2.0.0 for now

I've just tried it and I can confirm that things work well with v2.0.0. 

Thanks for the tip. Fingers crossed to have this issue fixed at some point.

**Comment by mathiasbynens** [comment]
#5654

**Comment by paullewis** [comment]
This is fixed as of 532ae573d29063ffd081ed9ee340d71b25320dec
